src/Apps/Artwork/ArtworkApp.tsx

```diff
@@ -230,6 +233,7 @@ export const ArtworkAppFragmentContainer = createFragmentContainer(
       ...ArtworkDetails_artwork
       ...ArtworkImageBrowser_artwork
       ...OtherWorks_artwork
+      ...PricingContext_artwork
```

Comment:
This winds up adding a potentially expensive stitched fetch (plus bandwidth) on all requests, even those that don't have the lab feature.

Can this be guarded with `...PricingContext_artwork @include(if: $includePricingFeature)`? That variable, `includePricingFeature`, can then be injected into the context by Force, which allows you to refer to it here.

Then, we can incrementally roll this out and test it w/o adding to load for everyone else. If/when the feature is rolled out to everyone it can just be referred to normally here.

LMK if you want some help setting that up! Here's some examples: https://github.com/artsy/force/blob/203673e3ce5c0c936d51ac3c66e19cbf471e8c70/src/desktop/apps/artist/server.tsx#L36-L39 on the server and https://github.com/artsy/force/blob/203673e3ce5c0c936d51ac3c66e19cbf471e8c70/src/desktop/apps/artist/client.tsx#L16-L19

Reply:
@mzikherman, good thinking!

Happily the way it's already implemented in metaphysics means that no extra fetching will be done for people who don't have the lab feature, so any perf impact should be negligible.

Reply:
I checked it out, and it looks like MP is fetching the user and doing a lab feature check: https://github.com/artsy/metaphysics/blob/9b5a3d892ac06f4e18223c3778cc64e8da144065/src/lib/stitching/vortex/stitching.ts#L93-L98

This is still a superfluous request. Force (and thus Reaction) _already_ has access to the current user serialized from the cookie, and so you can determine whether to fetch anything at this point...no need for that MP resolver to make a redundant request.

Reply:
Ah yeah, good catch! It's an extra fetch for logged-in users 🤔 I'll add a ticket to the purchase sprint. Thanks for raising this!